src/Symfony/Bundle/WebServerBundle/WebServerConfig.php

```diff
@@ -36,7 +36,11 @@ public function __construct($documentRoot, $env, $address = null, $router = null
 
         $this->documentRoot = $documentRoot;
         $this->env = $env;
-        $this->router = $router ?: __DIR__.'/Resources/router.php';
+        $this->router = $router ? realpath($router) : __DIR__.'/Resources/router.php';
+
+        if (false === $this->router || !file_exists($this->router)) {
```

Comment:
I think the file existence validation could be skipped  when using the built-in one. We already know it exists:

```php
if (null !== $router) {
    $absoluteRouterPath = realpath($router);

    if (false === $absoluteRouterPath || !file_exists($absoluteRouterPath)) {
        throw new \InvalidArgumentException(sprintf('Router script "%s" does not exists.', $router));
    }

    $this->router = $absoluteRouterPath;
} else {
    $this->router = __DIR__.'/Resources/router.php';
}